src/js/server/routes/apply.js

```diff
@@ -35,20 +38,143 @@ applyRouter.post('/', applyFormUpload.single('cv'), (req, res) => {
   form.handle(req.body, {
     success: () => {
       // redirect to the next page but for now...
-      res.redirect('/');
+      res.redirect('/form');
     },
     error: (resultForm) => {
-      renderApplyPageWithForm(res, resultForm);
+      renderApplyPageWithFormAndUser(res, resultForm, null);
     },
     empty: () => {
-      renderApplyPageWithForm(res, form);
+      renderApplyPageWithFormAndUser(res, form, null);
+    }
+  });
+});
+
+// The main apply page (has the login button)
+applyRouter.get('/', function (req, res) {
+  res.render('apply/index.html');
+});
+
+// The redirect to the authorization page
+applyRouter.get('/auth', function(req, res) {
+  let base_url = "https://my.mlh.io/oauth/authorize";
+
+  // Construct the query string
+  var qs = querystring.stringify({
+    client_id: client_id,
+    redirect_uri: auth_callback,
+    response_type: 'code',
+    scope: [
+      // All the user details we need
+      'email',
+      'phone_number',
+      'demographics',
+      'birthday',
+      'education',
+      'event'
+    ].join(' ')
+  });
+
+  // Redirect to the authorization page on MyMLH
+  res.redirect(base_url + "?" + qs);
+})
+
+// Redirect the request if there is no code in the query
+function ensureCode(req, res) {
+  if (req.query.code === undefined) {
+    res.redirect('/auth');
+  }
+}
+
+function getToken(code) {
+  // Now we have an authorization code, we can exchange for an access token
+  var base_url = "https://my.mlh.io/oauth/token";
+
+  // For debugging purposes
+  console.log(code);
+
+  var headers = { 'Content-Type': 'application/json' }
+  var body = {
+    'client_id': client_id,
+    'client_secret': client_secret,
+    'code': code,
+    'grant_type': "authorization_code",
+    'redirect_uri': auth_callback
+  }
+
+  return fetch(base_url, {
+
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    method: 'POST',
+    body: JSON.stringify(body)
+
+  }).then(function(response) {
+
+    return response.json();
+
+  }).then(function(json) {
+
+    return json.access_token;
+
+  });
```

Comment:
You've been hard at work! Good job on figuring out this MyMLH stuff.

There's quite a lot of logic in here. Do you think it would be better suited in its own `mlh-auth` module or something along those lines? It would help to keep our routes lean.